lib/configs/imports.ts

```diff
@@ -0,0 +1,127 @@
+import type { Linter } from 'eslint'
+/*!
+ * SPDX-FileCopyrightText: 2025 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+import type { ConfigOptions } from '../types.d.ts'
+
+import perfectionist from 'eslint-plugin-perfectionist'
+import {
+	GLOB_FILES_JAVASCRIPT,
+	GLOB_FILES_TYPESCRIPT,
+	GLOB_FILES_VUE,
+} from '../globs.ts'
+
+/**
+ * Generate imports and exports related ESLint rules.
+ *
+ * @param options - Configuration options
+ */
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export function imports(options: ConfigOptions): Linter.Config[] {
+	return [
+		{
+			name: 'nextcloud/imports/setup',
+			plugins: {
+				perfectionist,
+			},
+		},
+		{
+			name: 'nextcloud/imports/rules',
+			files: [
+				...GLOB_FILES_JAVASCRIPT,
+				...GLOB_FILES_TYPESCRIPT,
+				...GLOB_FILES_VUE,
+			],
+			rules: {
+				// Require file extensions
+				'no-restricted-imports': [
+					'error',
+					{
+						patterns: [
+							{
+								regex: '^(\\.*)/(.+/)*[^/.]+$',
+								message: 'Import is missing the file extension.',
+							},
+						],
+					},
+				],
+				// Sorting of imports
+				'sort-imports': 'off',
+				'perfectionist/sort-imports': [
+					'error',
+					{
+						type: 'natural',
+						newlinesBetween: 'never',
+						groups: [
+							// type first
+							'external-type',
+							'type',
+							{ newlinesBetween: 'always' },
+							// external things
+							[
+								'builtin',
+								'external',
+								'object',
+							],
+							// everything else which is everything internal
+							'unknown',
+							// Vue components
+							'vue', // external modules (e.g. nextcloud-vue)
```

Comment:
So we decided to go with this? external .vue components are now listed after internal non-vue modules
<img width="499" alt="image" src="https://github.com/user-attachments/assets/9f3aa2e1-7f92-4903-b35b-689ddbd22bb0" />

Reply:
@Antreesy one problem here is that you use the old imports they are not recognized as vue components. (The `/dist/Components`).
Instead use `/components/Nc..`.

Reply:
I would expect:

```
import type ...
import foo from 'bar'
import baz from './local'
import NcActions from '@nextcloud/vue...'
import SomeComponent from './SomeComponent.vue'

import 'other-side-effect'
```

Reply:
From the config it is:
- external
- internal
- vue external
- vue internal

Which is indeed different from the proposal above

Reply:
It was always like that here :eyes: 

- Typescript (external | internal)
- Javascript (external | internal)
- Vue (external | internal)
- Styles (external | internal)

Reply:
> Typescript (external | internal)\nJavascript (external | internal)

Does it actually separate TS from JS, or types from modules?

Reply:
> Otherwise why should you sort vue independently if you mix it with Javascript?

To not break it in the middle, because it can be in a single batch without mixing externals with internals

Reply:
> Does it actually separate TS from JS, or types from modules?

I do not mean `.ts` files I meat types.

- Types are stripped away so that block is only for type checking.
- Javascript is the code that stays so its the most important block.
- Vue files are components, you do not care in `<script>` about them so they do not need to interrupt reading the script sources. They are only relevant for the `<template>`.